llvm/lib/Transforms/Instrumentation/SPIRITTAnnotations.cpp

```diff
@@ -0,0 +1,294 @@
+constexpr char SPIRV_GROUP_FMAX[] = "GroupFMax";
+constexpr char SPIRV_GROUP_UMAX[] = "GroupUMax";
+constexpr char SPIRV_GROUP_SMAX[] = "GroupSMax";
+constexpr char SPIRV_ATOMIC_INST[] = "Atomic";
+constexpr char SPIRV_ATOMIC_LOAD[] = "AtomicLoad";
+constexpr char SPIRV_ATOMIC_STORE[] = "AtomicStore";
+constexpr char ITT_ANNOTATION_WI_START[] = "__itt_offload_wi_start_wrapper";
+constexpr char ITT_ANNOTATION_WI_RESUME[] = "__itt_offload_wi_resume_wrapper";
+constexpr char ITT_ANNOTATION_WI_FINISH[] = "__itt_offload_wi_finish_wrapper";
+constexpr char ITT_ANNOTATION_WG_BARRIER[] = "__itt_offload_wg_barrier_wrapper";
+constexpr char ITT_ANNOTATION_ATOMIC_START[] = "__itt_offload_atomic_op_start";
+constexpr char ITT_ANNOTATION_ATOMIC_FINISH[] =
+    "__itt_offload_atomic_op_finish";
+
+// Wrapper for the pass to make it working with the old pass manager
+class SPIRITTAnnotationsLegacyPass : public ModulePass {
+public:
+  static char ID;
+  SPIRITTAnnotationsLegacyPass() : ModulePass(ID) {
+    initializeSPIRITTAnnotationsLegacyPassPass(
+        *PassRegistry::getPassRegistry());
+  }
+
+  // run the SPIRITTAnnotations pass on the specified module
+  bool runOnModule(Module &M) override {
+    ModuleAnalysisManager MAM;
+    auto PA = Impl.run(M, MAM);
+    return !PA.areAllPreserved();
+  }
+
+private:
+  SPIRITTAnnotationsPass Impl;
+};
+
+} // namespace
+
+char SPIRITTAnnotationsLegacyPass::ID = 0;
+INITIALIZE_PASS(SPIRITTAnnotationsLegacyPass, "SPIRITTAnnotations",
+                "Insert ITT annotations in SPIR code", false, false)
+
+// Public interface to the SPIRITTAnnotationsPass.
+ModulePass *llvm::createSPIRITTAnnotationsPass() {
+  return new SPIRITTAnnotationsLegacyPass();
+}
+
+namespace {
+
+// Check for calling convention of a function.
+bool isSPIRKernel(Function &F) {
+  return F.getCallingConv() == CallingConv::SPIR_KERNEL;
+}
+
+Instruction *emitCall(Module &M, Type *RetTy, StringRef FunctionName,
+                      ArrayRef<Value *> Args, Instruction *InsertBefore) {
+  SmallVector<Type *, 8> ArgTys(Args.size());
+  for (unsigned I = 0; I < Args.size(); ++I)
+    ArgTys[I] = Args[I]->getType();
+  auto *FT = FunctionType::get(RetTy, ArgTys, false /*isVarArg*/);
+  FunctionCallee FC = M.getOrInsertFunction(FunctionName, FT);
+  assert(FC.getCallee() && "Instruction creation failed");
+  auto *Call = CallInst::Create(FT, FC.getCallee(), Args, "", InsertBefore);
+  return Call;
+}
+
+// Insert instrumental annotation calls, that has no arguments (for example
+// work items start/finish/resume and barrier annotation.
+bool insertSimpleInstrumentationCall(Module &M, StringRef Name,
+                                     Instruction *Position) {
+  Type *VoidTy = Type::getVoidTy(M.getContext());
+  ArrayRef<Value *> Args;
+  Instruction *InstrumentationCall = emitCall(M, VoidTy, Name, Args, Position);
+  assert(InstrumentationCall && "Instrumentation call creation failed");
+  return true;
```

Comment:
Do we really need to return that `true` here? Can we just make this function `void`?